File: leetcode/python/0009_palindrome_number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0: return False

        rev, org = 0, x
        while (x > 0):
            rev *= 10
            rev += x % 10
            x /= 10
        
        return rev == org

File: leetcode/cpp/0009_palindrome_number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0 || (x % 10 == 0 && x != 0)) {
            return false;
        }
        
        int reverseNum = 0, i = 1;
        while (x > reverseNum) {
            reverseNum = (reverseNum * 10) + x % 10;
            x /= 10;
        }
        
        if (reverseNum == x || reverseNum / 10 == x) {
            return true;
        }
        
        return false;
    }
};